src/webgl/p5.RendererGL.js

```diff
@@ -1096,7 +1099,7 @@ p5.RendererGL = class RendererGL extends p5.Renderer {
     else {
       pg.shader(this.filterShader);
       this.filterShader.setUniform('tex0', this);
-      this.filterShader.setUniform('texelSize', [1/this.width, 1/this.height]);
+      this.filterShader.setUniform('texelSize', texelSize);
       this.filterShader.setUniform('canvasSize', [this.width, this.height]);
```

Comment:
Should these values also be multiplied by pixel density?

Reply:
This is a good question. I think ideally, we'd want to make the shader handle pixel density rather than the user calling the filter function so that it's easier to make density-agnostic sketches. That means the shader would likely want to know about both the width and the pixel density. I think right now this is possible by comparing `texelSize` and `canvasSize`, but wouldn't be possible if we also multiply by pixel density here. (Maybe it's worth adding `pixelDensity` as a separate uniform so it doesn't always have to be calculated by dividing the other two?)

Reply:
I was thinking maybe all shader uniforms should account for pixel density, ie. `texelSize = 1 / (width * pixelDensity), 1 / (height * pixelDensity)` and `canvasSize = width * pixelDensity, height * pixelDensity`. That way they would all be doing work at the same level of resolution. Although having `canvasSize != width, height` is probably not what the user expects.